### ISSUE TITLE
Wrong return type on whereRaw

### ISSUE THREAD

**Issue by regnerisch** [nunomaduro/larastan#1956]
- Larastan Version: 2.9.6
- Laravel Version: 11.7.0
- Level: 9

### Description
When using `whereRaw` inside a `when` condition phpstan expects to return `Builder<Model>` not e.g. `Builder<User>`. Also I'm quite sure there is no need to return a value at all. 

### Laravel code where the issue was found

```php
        $subject = JobAdvertisement::query()
            ->where(fn (Builder $q) => $q
                ->when($request->str('radius'), function (Builder $q, Stringable $radius) use ($request) {
                    ['longitude' => $lng, 'latitude' => $lat] = $request->only('longitude', 'latitude');
                    $q->whereRaw("ST_Distance_Sphere(ST_GeomFromText('POINT($lng $lat)'), location_point) <= ?", [$radius->toString()]);
                })
            );
```

### Error
```
Anonymous function should return Illuminate\Database\Eloquent\Builder<Illuminate\Database\Eloquent\Model> but returns Illuminate\Database\Eloquent\Builder<Domain\JobAdvertisement\Models\JobAdvertisement>.  
         💡 Template type TModelClass on class Illuminate\Database\Eloquent\Builder is not covariant. Learn more: https://phpstan.org/blog/whats-up-with-template-covariant     
```

Adding `/** @var Builder<Model> $q */` and retuning the builder fixes the bug.

**Comment by calebdw** [comment]
Specifying `Builder` type in the closure will override/reset the generics---try removing them.

```php
$subject = JobAdvertisement::query()
    ->where(fn ($q) => $q
        ->when($request->str('radius'), function ($q, Stringable $radius) use ($request) {
            ['longitude' => $lng, 'latitude' => $lat] = $request->only('longitude', 'latitude');
            $q->whereRaw("ST_Distance_Sphere(ST_GeomFromText('POINT($lng $lat)'), location_point) <= ?", [$radius->toString()]);
    })
);

**Comment by regnerisch** [comment]
Seems to work. It's a little bit confusing to remove native Typehinting / -safty to ensure phpstan can detect things right.

**Comment by calebdw** [comment]
Looks like this might be fixed on [PHPStan v1.11.2](https://github.com/phpstan/phpstan/issues/7281)
